### PR TITLE
ENG_intern avoid regenerating makefile

### PR DESCRIPTION
**Pull request by KimlienHg**
Fastest hack to work around issues with native C extension in Jruby and permission. I generated the Makefile locally (with some modifications to install directories) and pushed it to the repo. `extconf.rb` is modified so that Makefile is not regenerated. As such this gem will no longer break `jruby -S bundle install`.

@jessica-muremure

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,8 @@
 .bundle
 .config
 .yardoc
+.gitignore
```
Comment (skrawcz) [inline .gitignore]: I think you want to remove that ;) (i.e. we want to track changes to .gitignore ;))

**Comment by skrawcz** [comment]
Can you elaborate a bit more in the commit how the hack is supposed to work? or in the files themselves? E.g. include any URLs you used to find what to do, etc please :)

**Comment by jessica-muremure** [comment]
Sorry I didn't see this earlier.  Let's go ahead and commit this.  I don't understand everything that's happening in your configuration, but it will be helpful to have this documented here, and I trust that it fixes the problems that you mention above.

File: .gitignore (hunk 1 above)
Comment (KimlienHg) [inline .gitignore]: definitely! huge oversight on my part -- I was shuffling things around and forgot to clean up.

**Comment by KimlienHg** [comment]
In the original gem, `extconf.rb` (most of which is now in `generate_makefile.rb`) uses [Ruby's Make Makefile module](https://github.com/ruby/ruby/blob/trunk/lib/mkmf.rb) to create a Makefile. This Makefile is called to compile and install the shared library `ext/cld/libcld2.so`.  
The process requires `ruby-dev` to be installed and admin privileges (and potentially other hurdles as well), which I don't have on Idibon servers. A quick solution is to push the locally compiled `libcld2.so` with the repo and modify `extconf.rb` to only generate Makefile when it is not already present (hack used by the [cld gem](https://github.com/jtoy/cld/tree/master/ext/cld)).  
This is definitely a very hacky solution --a ny other suggestion is welcomed. However since we want `cld2` as a makeshift tool to get language detection up and running, I'd recommend against spending too much time on this.

**Pull request merged**